README.md

```diff
@@ -1,44 +1,268 @@
 
-### Example Dockerfile
+A template is an easy starting point to build your own microservice. Its core purpose is to facilitate the development of a microservice that can run on the mu.semte.ch platform. There are templates available in several programming languages. As the name implies, this is a template to build microservices in Ruby.
 
-    FROM semtech/mu-ruby-template:2.12.0
-    LABEL maintainer="erika.pauwels@gmail.com"
-    # ONBUILD of mu-ruby-template takes care of everything
+#### Sinatra takes the stage
 
-### Versions
-The following versions of the mu-ruby-template are available:
-* `2.12.0`
-* `2.11.1`; `2.11.1-ruby2.5`
-* `2.10.0`; `2.10.0-ruby2.5`
-* `2.9.0` ; `2.9.0-ruby2.5`
-* `2.8.0` ; `2.8.0-ruby2.5`
-* `2.7.0` ; `2.7.0-ruby2.5`
-* `2.6.0` ; `2.6.0-ruby2.3`
-* `1.3.1-ruby2.1`
+The mu-ruby-template is based on the lightweight Ruby web framework [Sinatra](http://www.sinatrarb.com/). So in essence, everything you can do in Sinatra, you can also do in the template. We will start with the implementation of a simple ‘Hello World’ API endpoint.
 
-## Configuration
+Open a new file ‘web.rb’ and add the following lines of code:
+```rb
+get '/hello' do
+    status 200
+    { message: 'Hello world! }.to\_json
+end
+```
 
-The template supports the following environment variables:
+We’ve now defined a simple API endpoint on the path ‘/hello’. Next, we will wrap these lines of code in the mu-ruby-template.
 
-- `MU_SPARQL_ENDPOINT`: SPARQL read endpoint URL. Default: `http://database:8890/sparql` (the triple store should be linked as `database` to the microservice).
+Create a Dockerfile next to the web.rb file with the following content:
+```Dockerfile
+    FROM semtech/mu-ruby-template:2.4.0-ruby2.3
+```
+That’s it! You’ve build your first Ruby microservice in mu.semte.ch. If you now build this Docker image and include it in your mu-project, you will have a /hello endpoint available in your platform. Don’t forget to add a rule to your dispatcher to make this endpoint available to the frontend application.
 
-- `MU_APPLICATION_GRAPH`: configuration of the graph in the triple store the microservice will work in. Default: `http://mu.semte.ch/application`. The graph name can be used in the service via the `graph` helper method.
+#### Developing the microservice
 
-- `MU_SPARQL_TIMEOUT`: timeout (in seconds) for SPARQL queries. Default: 60 seconds.
+Building the Docker image each time you’ve changed some code is rather cumbersome during development. The template therefore supports development with live-reload. Start a container based on the mu-ruby-template image and mount your code in the /app folder:
+```bash
```

Comment:
I think a docker-compose snippet might be a useful addition here, as typically you'll run the service in a stack and not standalone